pkg/cache/clusterqueue.go

```diff
@@ -636,6 +636,24 @@ func (c *clusterQueue) isTASOnly() bool {
 	return true
 }
 
-func (c *clusterQueue) hasProvRequestAdmissionCheck() bool {
-	return len(c.provisioningAdmissionChecks) > 0
+func (c *clusterQueue) flavorsWithProvReqAdmissionCheck() sets.Set[kueue.ResourceFlavorReference] {
+	result := sets.New[kueue.ResourceFlavorReference]()
+	for _, ac := range c.provisioningAdmissionChecks {
+		result.Insert(c.flavorsForAdmissionCheck(ac).UnsortedList()...)
+	}
+	return result
+}
+
+func (c *clusterQueue) flavorsForAdmissionCheck(ac kueue.AdmissionCheckReference) sets.Set[kueue.ResourceFlavorReference] {
+	result := sets.New[kueue.ResourceFlavorReference]()
+	if len(c.AdmissionChecks[ac]) == 0 {
+		for _, rg := range c.ResourceGroups {
+			result.Insert(rg.Flavors...)
+		}
```

Comment:
Why do we iterate over resourceGroups if there is no flavor for admissionCheck. Is there an implicit assumption that c.AdmissionChecks[ac] returns only flavors that are indicated by AdmissionCheckStrategy?